daemon/src/taker_cfd.rs

```diff
@@ -167,6 +189,71 @@ impl<O, M, W> Actor<O, M, W> {
 
         Ok(())
     }
+
+    async fn handle_auto_rollover(&mut self) -> Result<()> {
+        tracing::debug!("Auto rollover");
+
+        let mut conn = self.db.acquire().await?;
+        let cfds = load_all_cfds(&mut conn).await?;
+
+        // cleanup all pending proposals because auto-rollover will create a new one
+        self.current_pending_proposals.clear();
+
+        // Only cfds in state `Open` are allowed for auto-rollover proposal
+        for cfd in cfds
+            .iter()
+            .filter(|cfd| matches!(cfd.state, CfdState::Open { .. }))
+        {
+            tracing::debug!(
+                "settlement_interval: {:?}",
+                self.roll_over_params.settlement_interval
+            );
+
+            let now = OffsetDateTime::now_utc();
+            let elapsed_since_creation = now - cfd.creation_timestamp()?;
+
+            // trigger rollover if rollover window has elapsed
+            if cfd.expiry_timestamp() > now
+                && elapsed_since_creation
+                    > self.roll_over_params.rollover_offset
```

Comment:
I cannot follow this logic, please explain what the values represent now and how they relate to each other.